highlight excluded nodes

lohathe opened the pull request.

lohathe committed:
Color GUI nodes whose names match user filters.

lohathe committed:
Add color info and percentage of exclusion to tree view. Add QSplitter as main widget.

lohathe commented:
Referring to #4

arl commented on backup-excluder.py:
```diff
@@ -2,23 +2,65 @@
+        theBrush = None
+        ratio = float(actualSize)/max([self.uncutSize, 1.0])
+        if exclusionState == SystemTreeNode.DIRECTLY_EXCLUDED:
+            theBrush = self.orangeBrush
```
Instead of setting a temporary variable `theBrush`, couldn't you directly call :

``` python
self._colorBackground(self.orangeBrush)
```

Even better, as it seems that for one _exclusion state_ constant we have one and only one _brush_, why not use a `dict`, in order to be able to do:

``` python
brushes = dict {
    SystemTreeNode.DIRECTLY_EXCLUDED: QBrush(QColor("orange")),
    SystemTreeNode.XXX_INCLUDED: QBrush(QColor("XXX")),
    # etc.
}
```

and then:

``` python
self._colorBackground(self.brushes[exclusionState])
```

arl commented on backup-excluder.py:
```diff
@@ -2,23 +2,65 @@
+            # Update the size of parents. Folder's size is 0 by default,
+            # this way it accounts for the files it contains.
+            sup = self.parent()
+            while sup and isinstance(sup, ExampleItem):
```
Isn't there another way to compute the directory size?

lohathe committed:
Refactor SystemTreeNode.createSystemTree.

Using recursion to create the SystemTreeNode tree, always based on os.fwalk.
The size of a inner node now remembers the size of the whole subtree rooted
in it.

lohathe committed:
Fixed SystemTreeNode.addChild.

Adding a new node now propagates its size to its ancestors.

lohathe committed:
Updated unittests, matching the refactored model.

lohathe committed:
Refactor brushes into a dictionary.

aurelien-rainone commented:
fixex #4
